lib/ws-server.js

```diff
@@ -34,6 +34,9 @@ function setupServer() {
 function handleConnect(ws) {
     _ws = ws;
     ws.on('message', handleMessage);
+    ws.on('close', function() {
+        events.emit('connection-closed');
+    });
```

Comment:
Could you move this into it's own named function? Trying to steer away from anon functions as much as possible (and keep away from the CB pyramid)